tests/tests/test-xcm/test-mock-hrmp.ts

```diff
@@ -1219,3 +1288,535 @@ describeDevMoonbeam("Mock XCM - receive horizontal transfer", (context) => {
+      WithdrawAsset: [
+        {
+          id: {
+            Concrete: {
+              parents: 0,
+              interior: {
+                X1: { PalletInstance: balancesPalletIndex },
+              },
+            },
+          },
+          fun: { Fungible: 1 },
+        },
+      ],
+    });
+
+    // we push the last buyExecution, that will fail
+    instructionsNotExecuted.push({
+      BuyExecution: {
+        fees: {
+          id: {
+            Concrete: {
+              parents: 0,
+              interior: {
+                X1: { PalletInstance: balancesPalletIndex },
+              },
+            },
+          },
+          fun: { Fungible: 1 },
+        },
+        weightLimit: { Limited: new BN(20000000000) },
+      },
+    });
+
+    instructionsExecuted.push({
+      WithdrawAsset: [
+        {
+          id: {
+            Concrete: {
+              parents: 0,
+              interior: {
+                X1: { PalletInstance: balancesPalletIndex },
+              },
+            },
+          },
+          fun: { Fungible: 2 },
+        },
+      ],
+    });
+
+    instructionsExecuted.push({
+      BuyExecution: {
+        fees: {
+          id: {
+            Concrete: {
+              parents: 0,
+              interior: {
+                X1: { PalletInstance: balancesPalletIndex },
+              },
+            },
+          },
+          fun: { Fungible: 2 },
+        },
+        weightLimit: { Limited: new BN(20000000000) },
+      },
+    });
+
+    let xcmMessageNotExecuted = {
+      V2: instructionsNotExecuted,
+    };
+    let xcmMessageExecuted = {
+      V2: instructionsExecuted,
+    };
+    const xcmpFormat: XcmpMessageFormat = context.polkadotApi.createType(
+      "XcmpMessageFormat",
+      "ConcatenatedVersionedXcm"
+    ) as any;
+    const receivedMessageNotExecuted: XcmVersionedXcm = context.polkadotApi.createType(
+      "XcmVersionedXcm",
+      xcmMessageNotExecuted
+    ) as any;
+
+    const receivedMessageExecuted: XcmVersionedXcm = context.polkadotApi.createType(
+      "XcmVersionedXcm",
+      xcmMessageExecuted
+    ) as any;
+
+    const totalMessageNotExecuted = [...xcmpFormat.toU8a(), ...receivedMessageNotExecuted.toU8a()];
+    const totalMessageExecuted = [...xcmpFormat.toU8a(), ...receivedMessageExecuted.toU8a()];
+
+    const paraId = context.polkadotApi.createType("ParaId", 1) as any;
+    const sovereignAddress = u8aToHex(
+      new Uint8Array([...new TextEncoder().encode("sibl"), ...paraId.toU8a()])
+    ).padEnd(42, "0");
+
+    // We first fund each sovereign account with 100
+    await context.createBlock(context.polkadotApi.tx.balances.transfer(sovereignAddress, 100n));
+
+    // now we start injecting messages
+    // two for para 1
+    await customWeb3Request(context.web3, "xcm_injectHrmpMessage", [1, totalMessageNotExecuted]);
+    await customWeb3Request(context.web3, "xcm_injectHrmpMessage", [1, totalMessageExecuted]);
+
+    await context.createBlock();
```

Comment:
`context.createBlock` accepts an array of transactions as well:

```suggestion
    await context.createBlock([
      customWeb3Request(context.web3, "xcm_injectHrmpMessage", [1, totalMessageNotExecuted]),
      customWeb3Request(context.web3, "xcm_injectHrmpMessage", [1, totalMessageExecuted]),
    ]);
```

Reply:
these are not transactions per se, so not sure how it will react to that

Reply:
oh yeah missed that, think `context.createBlock` only supports `eth_` web3requests as of now.